File: Client/src/app/@common/models/base-model.ts
export abstract class BaseModel {
  // ======================================= //
  public id         : number;
  public description: string;
  public title      : string;
  public created    : Date  ;
  // ======================================= //
  constructor(id: number, description: string, title: string, created: Date) {
    this.id          = id         ;
    this.description = description;
    this.title       = title      ;
    this.created     = created    ;
  }
  // ======================================= //
}

File: Shared/Models/BaseModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Shared.Models
{
    public abstract class BaseModel
    {
        // ======================================= //
        [Key]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime? Created { get; set; }
        // ======================================= //
        protected BaseModel()
        {
            this.Created = DateTime.Now;
        }
        protected BaseModel(string title, string description) : this()
        {
            this.Title = title;
            this.Description = description;
        }
    }
}